app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>CrudOperations</title>
+    <meta name="viewport" content="width=device-width,initial-scale=1">
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
```

Comment:
What are these 2 lines for?

Reply:
- `csrf_meta_tags` is added to prevent Cross-site Request Forgery (CSRF). The tag will return meta tags `csrf-param` and `csrf-token` with the name of the cross-site request forgery protection parameter and token, respectively.
-  `csp_meta_tag` is added to allow only trusted scripts in your app. It's used to protect your application from unknown scripts that could be injected to damage or hack your app.

Reply:
How does csrf_meta_tags prevent CSRF attacks?

Reply:
csrf_meta_tags embeds a unique CSRF token in the site's HTML. That same token is also stored in the session cookie. When a user makes a request, the token from the HTML gets sent with that request. Rails compares the token from the page with the token from the session cookie to ensure they match. If the token was not found within the request, or the value does not match, then the request should be aborted